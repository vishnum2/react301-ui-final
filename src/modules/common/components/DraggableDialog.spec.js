import React from 'react';
import DraggableDialog from './DraggableDialog';
import {shallow, mount} from '../../../enzyme';
import {PaperComponent} from './DraggableDialog';

describe('should check snapshot', () => {
    it('should check snapshot', () => {
        const wrapper = shallow(<DraggableDialog />);
        PaperComponent({});
        expect(wrapper).toMatchSnapshot();
    })
    
    it('simulate the dialog to open and test', async () => {
        const getMailMock = jest.fn();
        const email = 'vishnu.m2@mindtree.com';
        const wrapper = mount(<DraggableDialog getEmail={getMailMock} />);
        wrapper.setState({open: true});
        // console.log(wrapper.instance());
        wrapper.instance().handleOpenClose();
        expect(wrapper.state(['open'])).toEqual(false);
        wrapper.instance().handleSendAndClose(email);
        expect(getMailMock).toHaveBeenCalledWith(email);
        console.log(wrapper.debug());
    })

})