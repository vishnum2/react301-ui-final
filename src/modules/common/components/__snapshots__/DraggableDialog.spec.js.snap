// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should check snapshot should check snapshot 1`] = `ShallowWrapper {}`;
